specs/Workers.md

```diff
@@ -0,0 +1,1402 @@
+}
+
+[uuid(4e07e562-8db7-5815-907b-6d89a253a974), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerManager : IUnknown {
+  /// Adds an event handler for the `ServiceWorkerRegistered` event.
+  ///
+  /// A ServiceWorker is a specific type of worker that takes a JavaScript file
+  /// that can control the web-page/site that it is associated with,
+  /// intercepting and modifying navigation and resource requests, and caching
+  /// resources in a very granular fashion to give you complete control
+  /// over how app behaves in certain situations.
+  ///
+  /// Service workers essentially act as proxy servers that sit between web applications,
+  /// the browser, and the network (when available).
+  /// Unlike Shared Workers, which have their own separate global scope, Service
+  ///Workers have no DOM access and run in a different context.
+  ///
+  /// This event is raised when a web application registers a service worker using the
+  /// `navigator.serviceWorker.register("/sw.js")` method. See the
+  /// [Service Worker Registration](https://developer.mozilla.org/docs/Web/API/ServiceWorkerRegistration)
+  /// for more information.
+  ///
+  ///
+  /// \snippet ScenarioServiceWorkerManager.cpp ServiceWorkerRegistered
+  HRESULT add_ServiceWorkerRegistered(
+      [in] ICoreWebView2ServiceWorkerRegisteredEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Removes an event handler previously added with `add_ServiceWorkerRegistered`.
+  HRESULT remove_ServiceWorkerRegistered(
+      [in] EventRegistrationToken token);
+
+
+  /// Gets a list of the service worker registrations under the same profile.
+  ///
+  /// This method returns a list of `CoreWebView2ServiceWorkerRegistration` objects,
+  /// each representing a service worker registration.
+  ///
+  /// This method corresponds to the `getRegistrations` method of the `ServiceWorkerContainer`
+  /// object in the DOM which returns a Promise that resolves to an array of
+  /// `ServiceWorkerRegistration` objects. See the [MDN documentation]
+  /// (https://developer.mozilla.org/docs/Web/API/ServiceWorkerContainer/getRegistrations) for more information.
+  HRESULT GetServiceWorkerRegistrations(
+      [in] ICoreWebView2GetServiceWorkerRegistrationsCompletedHandler* handler
+  );
+
+  /// Gets the service worker registration object associated with the specified scope.
+  /// If a service worker has been registered for the given scope, it gets the
+  /// corresponding `CoreWebView2ServiceWorkerRegistration` object.
+  ///
+  /// The `scopeUri` parameter is a fully qualified URI that defines the range of URIs
+  /// a service worker can control. For example, if you registered a service worker
+  /// with a `scope` of '/app/' for an application at https://example.com/, you should
+  /// specify the full qualified URI i.e., https://example.com/app/ when calling this
```

Comment:
```suggestion
  /// a service worker can control. For example, instead of registering a service worker
  /// with a `scope` of '/app/' for an application at https://example.com/, you should
  /// specify the full qualified URI i.e., https://example.com/app/ when calling this
```
Clarify text. Original text was confusing, since it says "If you specified a scope of X, then you should specify a scope of Y", which is closing the barn door after the horse has bolted.

Reply:
I assume we return a failure if the scope is not fully qualified?

Reply:
Updated docs!